talisker/util.py

```diff
@@ -54,6 +54,25 @@ def parse_url(url, proto='http'):
     return urlparse(url)
 
 
+CLEANED = '{scheme}://{user}{colon}{password}{at}{hostname}{port}{path}{qs}'
+
+
+def sanitize_url(url):
+    """Strips a url of any pw and query strings."""
+    parsed = parse_url(url)
+    return CLEANED.format(
+        scheme=parsed.scheme,
+        user=parsed.username or '',
+        colon=':' if parsed.password else '',
+        password='*' * len(parsed.password) if parsed.password else '',
```

Comment:
do we really want to leak len(password) ? I'd suggest hardcoding eight asterisks

Reply:
ok